pkg/asset/machines/master.go

```diff
@@ -41,6 +41,10 @@ func (m *Master) Name() string {
 func (m *Master) Dependencies() []asset.Asset {
 	return []asset.Asset{
 		&installconfig.ClusterID{},
+		// PlatformCredsCheck just checks the creds (and asks, if needed)
+		// We do not actually use it in this asset directly, hence
+		// it is put in the dependencies but not fetched in Generate
+		&installconfig.PlatformCredsCheck{},
```

Comment:
We don't need this anymore with 727123c46f do we?  Or the similar `Worker` change?  `GetSession` (and similar for OpenStack, etc.) will handle it internally if/when these functions need to lookup credentials.  And the `Cluster` dependency backstops you from "all my assets were generated somewhere else, but now I'm calling `create cluster` without credentials.

Reply:
> We don't need this anymore with 727123c do we?

Ah, we will want these if we go with @staebler's [plan][1] to pass the session around as structured asset data, vs. my recreating a session whenever you need it.  Once session instance is a bit cleaner, so don't worry about me here ;).

[1]: https://github.com/openshift/installer/pull/1100#issuecomment-457768892

Reply:
Actually, this should be the preferred way I think, because only the 'asset' should ideally ask user any questions (when creds are missing). Commit 727123c is more of streamlining it. I would agree with @staebler's [comment](https://github.com/openshift/installer/pull/1100#issuecomment-457768892) here.